fw/http_parser.c

```diff
@@ -3635,10 +3635,17 @@ tfw_http_parse_req(void *req_data, unsigned char *data, size_t len,
 	/* ----------------    Request Line    ---------------- */
 
 	/* Parser internal initializers, must be called once per message. */
-	__FSM_STATE(Req_0, hot) {
-		if (unlikely(IS_CRLF(c)))
-			__FSM_MOVE_nofixup(Req_0);
-		/* fall through */
+	__FSM_STATE(Req_0_CR, hot) {
+		if (unlikely(c == \r))
+			__FSM_MOVE_nofixup(Req_0_LF);
+	}
+	__FSM_STATE(Req_0_LF, hot) {
+		if (unlikely(c == \n))
+			__FSM_MOVE_nofixup(Req_0_ExtraLeadingCRLF);
+	}
+	__FSM_STATE(Req_0_ExtraLeadingCRLF, hot) {
+		if (unlikely(c <= 0x20))
```

Comment:
Why to compare it with 0x20? If we have 0xFF - is it better, than any character below 0x20?

Reply:
@krizhanovsky 
<= 0x20 covers \r \n \t, blank, and a set of unusual characters.
Of course, we could make check for readably ASCII and even exactly for characters from which HTTP method can start.
Hower, this will be longer and will partialy duplicate subsequent method parsing.

Reply:
Well, most of the parsers (e.g. Cloudflare) do the checks like this, but we do accurate checking and do this in a fast way https://habr.com/ru/company/oleg-bunin/blog/493242/

Reply:
@Dmitry-Gouriev I noticed that the parser code is gone now, probably mistakenly. We DO NEED the parser changes to block unlimited number of preceeding CRLFs.